src/openms/include/OpenMS/ANALYSIS/OPENSWATH/DIAScoring.h

```diff
@@ -136,6 +136,9 @@ namespace OpenMS
     /// Precursor isotope scores for precursors (peptides and metabolites)
     void dia_ms1_isotope_scores(double precursor_mz, SpectrumPtrType spectrum, size_t charge_state, 
                                 double& isotope_corr, double& isotope_overlap, const std::string& sum_formula = "") const;
+    void dia_ms1_isotope_scores(double precursor_mz, SpectrumPtrType spectrum,
```

Comment:
this looks redundant
move charge_state e.g., before sum_formula and add a default value?
would make for a slimmer interface

Reply:
I am not sure about the default charge state value, since for peptides it would most probably +2 (I guess). For metabolites, it would be either +1 (positive mode) or -1 (negative mode)

Reply:
I can move the param, but I dont think it is redundant right? Since sum_formula can be empty, we need to require it. I have the new overload that is slimmer.

Reply:
Hmm ok then charge_state should be an int. How did you do it currently @oliveralka :)

Reply:
Usually in my case charge is already set by SIRIUS. It also can be assessed the annotated adduct, which currently only covers charge +1/-1.
https://github.com/OpenMS/OpenMS/pull/4725/files/14f627ec39174b4e499534a241465e1f9fc905dc#diff-d315275bb3622b58d06fd4ea2199d75255d7338e2f1aee06f421ea27bdad0d20R49

If neither sumformula nor adducts are set you would probably assume a positive/negative charge based on the polarity of the experiment.

Reply:
No, I mean in the current state OpenSwath will evaluate this score with a charge of 2^32 - 1 since the integer -1 is converted to an unsigned int.